src/fairseq2/metrics/recorder.py

```diff
@@ -68,7 +68,11 @@ def format_as_float(value: Any, *, postfix: Optional[str] = None) -> str:
 
 
 def register_metric_formatter(
-    name: str, display_name: str, priority: int, format_fn: Callable[[Any], str]
+    name: str,
+    display_name: str,
+    priority: int,
+    format_fn: Callable[[Any], str],
+    overwrite: Optional[bool] = False,
```

Comment:
Nit: No need for `Optional`. This can be just `overwrite: bool = False`.

Reply:
Can you also write a short doctstr for `overwrite`? Something like `If ``True``, overwrites any existing metric with the same name.`.